src/AI/Muriel.java

```diff
@@ -794,34 +796,39 @@ private void queueUnitProductionActions(GameMap gameMap)
       }
 
       // Grab the first enemy unit type, and try to build something that will counter it.
-      UnitModel enemyToCounter = enemyUnitStrengths.peek().model;
+      ModelForCO enemyToCounter = enemyUnitStrengths.peek().model;
       log("Want to counter " + enemyToCounter);
       log(String.format("  Remaining budget: %s", budget));
 
       // If we have a lot of cash on hand, don't worry about cost effectiveness - just maximize damage instead.
       // If we ever collect more than twice our income in funds, we just aren't spending fast enough. Fix that.
-      int incomePerTurn = myCo.getIncomePerTurn();
-      boolean useDamageRatio = (myCo.money > (incomePerTurn*2)); // Rich people can afford to think differently.
+      int incomePerTurn = myArmy.getIncomePerTurn();
+      boolean useDamageRatio = (myArmy.money > (incomePerTurn*2)); // Rich people can afford to think differently.
       if(useDamageRatio) log("  High funds - sorting units by damage ratio instead of cost effectiveness.");
 
       // If we are low on grunts, make sure we save money to build more.
-      UnitModel infModel = myCo.getUnitModel(UnitModel.TROOP);
+      UnitModel infModel = myArmy.cos[0].getUnitModel(UnitModel.TROOP);
+      int infCount = 0;
+      for( Unit u : myArmy.getUnits() )
+        if( infModel == u.model )
+          ++infCount;
```

Comment:
Why not use `myUnitCounts` to count the number of infantry?

Reply:
There's a relevant semantic difference:
Just counting the inf gives you the number of units.
`myUnitCounts` reckons injured units as fractional units, which may or may not be as relevant for infantry.

Reply:
This changes Muriel's behavior though, which is def out of scope for this PR. Why did this need to change?

Reply:
Oh. My brain decided at some point that this is what Muriel did before, and therefore I was doing this to maintain the old function.
Sorry for the confusion.